block/manager.go

```diff
@@ -417,6 +417,11 @@ func (m *Manager) BlockStoreRetrieveLoop(ctx context.Context) {
 		select {
 		case <-waitCh:
 			for {
+				select {
+				case <-ctx.Done():
+					return
```

Comment:
So is the bug just not respecting context? 
I wonder why the called func below didn't return

Reply:
without
```
select {
case <-ctx.Done():
     return
default:
}
```
the inner for-loop would never exit

Reply:
Removed the loop entirely since there is already an existing outer for loop with a select that should handle everything.